bluebottle/statistics/views.py

```diff
@@ -1,15 +1,129 @@
+
+from django.db import connection
+from django.db.models.aggregates import Sum
+from django.core.cache import cache
+
+from bluebottle.utils.utils import StatusDefinition
+
 from .serializers import StatisticSerializer
 
 
+class Statistics(object):
+    def _get_cached(self, key):
+        tenant_name = connection.tenant.client_name
+        tenant_key = '-'.join([tenant_name, key])
+
+        return cache.get(tenant_key)
+
+    def _set_cached(self, key, value, timeout=300):
+        tenant_name = connection.tenant.client_name
+        tenant_key = '-'.join([tenant_name, key])
+
+        return cache.set(tenant_key, value, timeout)
+
+    def clear_cached(self):
+        tenant_name = connection.tenant.client_name
+
+        for key in ['people-involved-total', 'projects-realized-total',
+                    'projects-online-total', 'donations-total', 'tasks-realized-total']:
+            tenant_key = '-'.join([tenant_name, key])
+            cache.set(tenant_key, None, 0)
+
+    @property
+    def people_involved(self):
+        from bluebottle.tasks.models import TaskMember, Task
+        from bluebottle.orders.models import Order
+        from bluebottle.projects.models import Project
+        from bluebottle.fundraisers.models import Fundraiser
+        from bluebottle.members.models import Member
+
+        """
+        Count all people who donated, fundraised, campaigned or was
+        a task member. People should be unique across all categories.
+        """
+        if False and self._get_cached('people-involved-total'):
+            return self._get_cached('people-involved-total')
+
+        donator_ids = Order.objects.filter(status__in=(StatusDefinition.PENDING, StatusDefinition.SUCCESS)).order_by('user__id').distinct('user').values_list('user_id', flat=True)
+        fundraiser_owner_ids = Fundraiser.objects.order_by('owner__id').distinct('owner').values_list('owner_id', flat=True)
+        project_owner_ids = Project.objects.filter(status__slug__in=('campaign', 'done-complete', 'done-incomplete',)).order_by('owner__id').distinct('owner').values_list('owner_id', flat=True)
+        task_member_ids = TaskMember.objects.order_by('member__id').distinct('member').values_list('member_id', flat=True)
+        task_owner_ids = Task.objects.order_by('author__id').distinct('author').values_list('author_id', flat=True)
+
+        items = [donator_ids, fundraiser_owner_ids, project_owner_ids, task_member_ids, task_owner_ids]
+
+        # get count of unique member ids
+        seen = set()
+        seen_add = seen.add
+        people_count = len([item for item in list(itertools.chain(*items)) if item and not (item in seen or seen_add(item))])
+
+        # Add anonymous donators
+        people_count += Order.objects.filter(user_id=None, status__in=(StatusDefinition.PENDING, StatusDefinition.SUCCESS)).count()
+
+        # Add "plus one"
+        people_count += TaskMember.objects.all().aggregate(externals=Sum('externals'))['externals'] or 0
```

Comment:
This is actually the line that fixes the bug, and adds all the "plus ones" to the stats. the rest is a refactor.